lib/src/gregorian_calendar_duration.dart

```diff
@@ -0,0 +1,157 @@
+import 'calendar_duration.dart';
+import 'gregorian_calendar.dart';
+import 'gregorian_consts.dart';
+
+/// Represents a given duration which can be added/subtracted
+/// from a [GregorianCalendar] object
+class GregorianCalendarDuration implements CalendarDuration<GregorianCalendar>{
+
+  GregorianCalendarDuration({this.days = 0, this.weeks = 0, this.months = 0, this.years = 0});
+
+  /// Constructs a GregorianCalendarDuration where the values have been normalized
+  factory GregorianCalendarDuration.normalized({int years = 0, int months = 0, int weeks = 0, int days=0}){
+    // normalize values and then call the real constructor
+    if (days >= daysPerWeek){
+      weeks += days ~/ daysPerWeek;
+      days %= daysPerWeek;
+    }
+    else if (days <= -daysPerWeek){
+      weeks += days ~/ daysPerWeek;
+      days = -(days.abs() % daysPerWeek);
+    }
+    if (months >= monthsPerYear){
+      years += months ~/ monthsPerYear;
+      months %= monthsPerYear;
+    }
+    else if (months <= -monthsPerYear){
+      years += months ~/ monthsPerYear;
+      months = -(months.abs() % monthsPerYear);
+    }
+    return GregorianCalendarDuration(years: years, months: months, weeks: weeks, days: days);
+  }
+
+  factory GregorianCalendarDuration.fromDays(int days, GregorianCalendar referenceDate){
+    final newReferenceDate = referenceDate.addDays(days);
+    return new GregorianCalendarDuration.normalized(
+      years: newReferenceDate.year - referenceDate.year,
+      months: newReferenceDate.month - referenceDate.month,
+      days:  newReferenceDate.day - referenceDate.day
+    );
+  }
+
+
+
+
+  @override
+  final int days;
+
+  @override
+  final int weeks;
+
+  @override
+  final int months;
+
+  @override
+  final int years;
+
+
+
+  @override
+  GregorianCalendarDuration operator +(CalendarDuration<GregorianCalendar> other){
+    return new GregorianCalendarDuration.normalized(
+      years: years + other.years,
+      months: months + other.months,
+      weeks: weeks + other.weeks,
+      days: days + other.days,
+    );
+  }
+
+  @override
+  GregorianCalendarDuration operator -(CalendarDuration<GregorianCalendar> other){
+    return new GregorianCalendarDuration.normalized(
+      years: years - other.years,
+      months: months - other.months,
+      weeks: weeks - other.weeks,
+      days: days - other.days,
+    );
+  }
+
+  @override
+  GregorianCalendarDuration operator -(){
+    return new GregorianCalendarDuration(
+      years: -years,
+      months: -months,
+      weeks: -weeks,
+      days: -days,
+    );
+  }
+
+
+  @override
+  num toComparableNum(){
+    return (years * avgDaysPerYear) + (months * avgDaysPerMonth) + (weeks * daysPerWeek) + days;
```

Comment:
This will be prone to precision errors, no?

Reply:
Yes, there would be precision errors involved. I'm not sure if there is really an excellent way to accomplish comparisons since in the real world years and months don't divide nice and evenly into days. I feel like it's probably good enough for most purposes though. In situations requiring exact precision, the user should select a reference date and use `CalendarDuration.toDays(referenceDate)`. 

We could probably improve consistency a little by doing `(years * monthsPerYear + months) * avgDaysPerMonth`. That would at least guarantee that 1 year always exactly equals 12 months.

I guess the other option we could do is have a hard-coded reference date to perform comparisons against. I don't really like that idea because it seems like a lot of extra processing overhead and code complexity, plus doing that would lead to inconsistent-feeling behavior such as 1 month > 30 days but 2 months < 60 days (assuming the reference date was January 1st of a non-leap year)

Reply:
Another thought would be to do away with truncate the result to an integer and rename the method to `toDaysEstimated` to indicate it's a less precise version of `toDays`

Reply:
Method removed in recent commits.